tests/test_display.py

```diff
@@ -59,6 +63,83 @@ def test_display(self, mock_put_text):
         mock_put_text.assert_called_with(self.img, 'Count: 10', (360, 60))
 
 
+class TestDisplayClassnameOverlay(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.img = Mock()
+        self.img.shape = (510, 640, 3)
+        self.font_scale = 3.0
+        self.thickness = 2
+        self.duration = 2
+        self.half_duration = self.duration / 2
+
+    @patch('sense.display.put_text')
+    def test_display_below_threshold(self, mock_put_text):
+        test_display = base_display.DisplayClassnameOverlay(thresholds={'Dabbing': 0.5})
+        test_display.display(self.img, {'sorted_predictions': [['Dabbing', 0.4]]})
+
+        mock_put_text.assert_not_called()
+
+    @patch('sense.display.put_text')
+    def test_display_class_not_in_threshold(self, mock_put_text):
+        test_display = base_display.DisplayClassnameOverlay(thresholds={'Dabbing': 0.5})
+        test_display.display(self.img, {'sorted_predictions': [['Nodding', 1.0]]})
+
+        mock_put_text.assert_not_called()
+
+    @patch('sense.display.put_text')
+    def test_display_default(self, mock_put_text):
+        test_display = base_display.DisplayClassnameOverlay(thresholds={'Dabbing': 0.5})
+        test_display.display(self.img, {'sorted_predictions': [['Dabbing', 0.6]]})
+
+        mock_put_text.assert_called_with(self.img,
+                                         'Dabbing',
+                                         font_scale=self.font_scale,
+                                         position=(224, 294),
+                                         thickness=self.thickness)
+
+    @patch('sense.display.DisplayClassnameOverlay._display_class_name')
+    def test_display_default_within_duration(self, mock_display_class_name):
+        test_display = base_display.DisplayClassnameOverlay(thresholds={'Dabbing': 0.5},
+                                                            duration=self.duration)
+
+        # Set start time and call display once
+        test_display._start_time = test_display._get_current_time()
+        test_display.display(self.img, {'sorted_predictions': [['Dabbing', 0.51]]})
+        mock_display_class_name.assert_called_with(self.img, 'Dabbing')
+        assert mock_display_class_name.call_count == 1
+
+        # Call display once during the duration
+        with patch("sense.display.DisplayClassnameOverlay._get_current_time",
+                   return_value=test_display._start_time + self.half_duration):
+            test_display.display(self.img, {'sorted_predictions': [['Dabbing', 0.51]]})
+
+        # Call display post the duration
+        with patch("sense.display.DisplayClassnameOverlay._get_current_time",
+                   return_value=test_display._start_time + self.duration):
+            # Display nothing after duration has passed
+            test_display.display(self.img, {'sorted_predictions': []})
+
+        # Total call counts should remain the same
+        assert mock_display_class_name.call_count == 2
```

Comment:
Theoretically it's not clear, where your `call_count` has increased. Could have happened after the first or second call.